### DISCUSSION TITLE
纯求知: Custome Element里slot的内容会在shadowRoot下以及shadowRoot旁边各出现一份?

### DISCUSSION THREAD

**Discussion by yoyo837** [hellof2e/quark-design#273] [Q&A] [answered]
这是符合预期的吗? 还是小弟哪里搞的不对? 个人不太懂这个机制.

**Comment by xxxsf** [comment]
相当于一个软链，使用了slot，就会出现在shadowRoot同级

**Reply by yoyo837** [reply]
<img width="699" alt="image" src="https://github.com/hellof2e/quark-design/assets/6134547/2e6eb694-180c-4ec3-94b4-bc5ef23ff77f">

可以帮我看看我编译时哪里处理的不对吗?

**Reply by yoyo837** [reply]
上面这个是我编译的, 下面这个是直接用quark-button的

**Reply by yoyo837** [reply]
如果我放的slot是一个某实例的元素, 这样会有两个实例, 我如何让它只有一个实例?

**Reply by xxxsf** [reply]
用的是quarkc写的吗？

**Reply by yoyo837** [reply]
不是, 是vue-cli编译的, 它编译的没有`reveal`, 是因为这个造成的多实例吗?不行我就切到 quarkc来了

**Reply by xxxsf** [reply]
用quarkc写吧，你用vue-cli编译，就违背  WCs 设计初衷了

**Reply by xxxsf** [reply]
quarkc 编译出来才是最“原生”的

**Reply by yoyo837** [reply]
图快直接问一下, `quarkc`功能是完备的了吗?

**Reply by sanqi-med** [reply]
完备， quarkd 就是用 quarkc 编译的，quarkd 目前已服务集团所有 h5 应用两年多

**Reply by yoyo837** [reply]
好的, 我晚点研究下

**Reply by xxxsf** [reply]
既然想用WCs，就是要让组件“远离”框架，你怎么还会去用vue呢。这个组件开发”原则“坚守一下啊，哈哈哈😂

**Reply by yoyo837** [reply]
我一开始没发现`quarkc`, 只主意到了`quarkd`

**Reply by yoyo837** [reply]
后来基于`vue-cli`写了一些基础后就暂时没动它, 直到发现title这个问题.

**Reply by yoyo837** [reply]
我在vue那边基于一些hack方式实现了WCs之间的响应式的provide、inject, 请问quarkc这边对此有推荐做法吗?

**Reply by xxxsf** [reply]
vue 不太了解，写的也不多。有quarkc相关问题可以直接问。

**Reply by yoyo837** [reply]
想要实现WCs之间的响应式的provide、inject, 或者向上取最近的xxx元素(自定义元素)的实例或者数据, 请问quarkc这边对此有推荐做法吗?

**Reply by sanqi-med** [reply, marked as answer]
quarkc 目前的玩法是类似原生标签，组件的通信主要是通过事件，如果想在父组件里面拿子组件的数据，通过事件即可（或者是父组件里面拿子组件的实例，然后调用实例方法拿数据，不推荐）。如果想在子组件拿父组件的数据通过props传递即可。如果是应用层面的公共数据想在自定义组件里面获取，可以使用 store 类似的机制，存内层里的store，或者浏览器的store都可以达到你的目的。

**Reply by yoyo837** [reply]
感谢各位热心帮助.
